### PR TITLE
Handle `balanceError` case

### LINKED ISSUE #2082: Mobile Error : Mobile client does not recognize balance of several ERC20 tokens. Forced to reinstall
**Describe the bug**
_Metamask on mobile client (IOS) does not recognize balance when new token is added from list. Refreshing does not work, bug is persistent. Have not attempted to reinstall app. Browser client and etherscan both recognize balances. When attempting to send funds 'Undefined Balance' error is prompted._

**Screenshots**
Version:
IMG_3964

Etherscan:
IMG_3965

Mobile Client:
IMG_3963

**To Reproduce**
_Steps to reproduce the behavior_
1. Send token or swap on DEX
2. Add token to metamask 'List' on mobile
3. Attempt to use tokens

**Smartphone (please complete the following information):**
 - Device: [iPhone 11 Pro]
 - OS: [e.g. iOS 14.2]
 - App Version [e.g. v1.0.8 (554)]

-------------------------------------------------------------
_to be added after bug submission by internal support / PM_
**Severity**

- How critical is the impact of this bug on a user?
- Add stats if available on % of customers impacted
- Is this visible to all users?
- Is this tech debt?

### LINKED ISSUE #2128: Bug: token balance shows 0 if one of the tokens' property fails
### Problem
This is an issue with error handling. When a token (like synthetix, sUSD) fails to render, all other token balances render as 0. This is very confusing, we've seen some support tickets come in for this. 

Workaround not _really_ available: Usually if the user hides the token that is causing the issue (no way to know for them, unless our support team guides them). This workaround isn't scalable. User isn't communicated what is failing and why this might be happening. Even if they reinstall or reimport their wallet, the problem persists.

We have see instances on both iOS and Android

Support tickets for more context:
https://consensys.zendesk.com/agent/tickets/76825
https://consensys.zendesk.com/agent/tickets/76776
https://consensys.zendesk.com/agent/tickets/77294
https://consensys.zendesk.com/agent/tickets/75337

### PR DESCRIPTION
**Pull request by rickycodes**
**Description**

<strike>This is WIP since it relies on https://github.com/MetaMask/controllers/pull/328</strike>

Now when a token has a balance error we display it in the UI:

image

And then on the token itself we display a warning:

image

**Checklist**

* [x] There is a related GitHub issue
* [x] Tests are included if applicable
* [x] Any added code is fully documented

**Issue**

Resolves #2128

### CODE REVIEW

File: app/components/UI/AssetOverview/index.js [hunk 1]
```diff
@@ -139,10 +159,28 @@ class AssetOverview extends PureComponent {
 		);
 	};
 
+	renderWarning = () => {
+		const {
+			asset: { symbol }
+		} = this.props;
+		// const etherscanUrl = `https://etherscan.io/token/${address}?a=${selectedAddress}`;
```
Comment (estebanmino) [inline app/components/UI/AssetOverview/index.js:166]: do we need this?

File: package.json [hunk 2]
```diff
@@ -74,7 +74,7 @@
     "@estebanmino/controllers": "3.2.38",
     "@exodus/react-native-payments": "https://github.com/wachunei/react-native-payments.git#package-json-hack",
     "@metamask/contract-metadata": "^1.22.0",
-    "@metamask/controllers": "^6.0.1",
+    "@metamask/controllers": "./metamask-controllers-v6.0.0.tgz",
```
Comment (estebanmino) [inline package.json:77]: this need an update 😅

File: app/components/UI/AssetOverview/index.js [hunk 3]
```diff
@@ -175,20 +213,28 @@ class AssetOverview extends PureComponent {
...
-					middleText={strings('asset_overview.receive_button').toUpperCase()}
-					onLeftPress={this.onSend}
-					onMiddlePress={this.onReceive}
-					middleType={'receive'}
-				/>
+				{!balanceError && (
+					<AssetActionButtons
+						leftText={strings('asset_overview.send_button').toUpperCase()}
+						testID={'token-send-button'}
+						middleText={strings('asset_overview.receive_button').toUpperCase()}
```
Comment (estebanmino) [inline app/components/UI/AssetOverview/index.js:232]: do we have this uppercased?

File: app/components/UI/AssetOverview/index.js (hunk 1 above)
Comment (rickycodes) [reply app/components/UI/AssetOverview/index.js:166]: I can remove this comment. vestigial leftovers from something that was there previously :laughing:

File: package.json (hunk 2 above)
Comment (rickycodes) [reply package.json:77]: this is just for QA so @ibrahimtaveras00 can see a token failure (I'll be reverting this before merge)

File: app/components/UI/AssetOverview/index.js (hunk 3 above)
Comment (rickycodes) [reply app/components/UI/AssetOverview/index.js:232]: I didn't add these `toUpperCase()` calls - only the `balanceError` check around them

**Review by ibrahimtaveras00 — APPROVED** [review]
👍🏽

**Review by estebanmino — APPROVED** [review]
👌

**Pull request merged**